src/extensionFunctions.ts

```diff
@@ -896,4 +899,41 @@ export function sendTextToNamedOutputChannel(message: string, channelName?: stri
 	if (!channelName || channelName === DIDACT_OUTPUT_CHANNEL) {
 		didactOutputChannel.show();
 	}
-}
\ No newline at end of file
+}
+
+export async function copyFileTextToClipboard(uri: vscode.Uri) : Promise<void> {
+	let testForPrefix : string = uri.toString();
+	let testUri : vscode.Uri = uri;
+	if (testForPrefix && !testForPrefix.startsWith('didact://?')) {
+		testForPrefix = 'didact://?&' + testForPrefix; // preface so we can use the existing method to parse for path
+		testUri = vscode.Uri.parse(testForPrefix);
+	}
+	const out : vscode.Uri | undefined = handleVSCodeDidactUriParsingForPath(testUri);
+	if (!out) {
+		const errmsg = `ERROR: No file found when parsing path ${uri}`;
+		sendTextToOutputChannel(errmsg);
+		vscode.window.showErrorMessage(errmsg);
+		throw new Error(errmsg);
+	} else {
+		let content : string | undefined = undefined;
+		if (out.scheme === 'file') {
+			await getDataFromFile(out)
```

Comment:
what about using directly` fs.readFileSync(uri.fsPath, 'utf8');`?
it will:
- avoid to reuse the getDatafromFile which is doing a transformation for Didact files and not for this case. (leading to think that the getDataFormFile might be better named to have the information that it is providing the transformed value of adoc/md files but better in another PR)
- allows to copy raw data from a `adoc` or `md` file
- avoid extra filename check